core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -990,12 +1005,23 @@ object Parser {
     * note: this can wind up parsing nothing
     */
   def repAs0[A, B](p1: Parser[A])(implicit acc: Accumulator0[A, B]): Parser0[B] =
-    Impl.Rep0(p1, acc)
+    Impl.Rep0(p1, Int.MaxValue, acc)
+
+  /** Repeat this parser 0 or more times
+    * note: this can wind up parsing nothing
+    */
+  def repAs0[A, B](p1: Parser[A], max: Int)(implicit acc: Accumulator0[A, B]): Parser0[B] =
+    Impl.Rep0(p1, max - 1, acc)
 
   /** Repeat this parser 1 or more times
     */
   def repAs[A, B](p1: Parser[A], min: Int)(implicit acc: Accumulator[A, B]): Parser[B] =
-    Impl.Rep(p1, min, acc)
+    Impl.Rep(p1, min, Int.MaxValue, acc)
+
+  /** Repeat this parser 1 or more times
+    */
+  def repAs[A, B](p1: Parser[A], min: Int, max: Int)(implicit acc: Accumulator[A, B]): Parser[B] =
+    Impl.Rep(p1, min, max - 1, acc)
```

Comment:
Either we should asset that min <= max or r we should return fail in the case max < min

What do you think? I tend towards error since that won't be recovered at parse time with a oneOf/orElse.

Reply:
Let's go with the error at parser creation. If we return fail, there is the next sticky dilemma of whether it should consume up until `max` repetitions before failing or not. Just making it an error side-steps all that.